### PR TITLE
Zuul push oss

### PR DESCRIPTION
**Pull request by raksoras**
Open sourcing Zuul Push (WebSocket/SSE) implementation.

### CODE REVIEW

File: zuul-core/src/main/java/com/netflix/zuul/netty/server/push/PushAuthHandler.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            }
+        }
+        else {
+            sendHttpResponse(req, ctx, NOT_FOUND);
+        }
+    }
+
+    protected final Cookies parseCookies(FullHttpRequest req) {
+        final Cookies cookies = new Cookies();
+        final String cookieStr = req.headers().get(HttpHeaderNames.COOKIE);
```
Comment (sudheerv) [inline zuul-core/src/main/java/com/netflix/zuul/netty/server/push/PushAuthHandler.java]: Hmm..doesn't this return a list when there are multiple Cookies involved?

**Comment by sudheerv** [comment]
changes look good to me 👍

File: zuul-core/src/main/java/com/netflix/zuul/netty/server/push/PushAuthHandler.java (hunk 1 above)
Comment (raksoras) [reply zuul-core/src/main/java/com/netflix/zuul/netty/server/push/PushAuthHandler.java]: I believe SET-COOKIE and be multiple but when user agent sends cookies back it is supposed to merge them all in single COOKIE header? This is what is running in production today...

Comment (sudheerv) [reply zuul-core/src/main/java/com/netflix/zuul/netty/server/push/PushAuthHandler.java]: Hmm..yeah, it's a bit unclear..most of the other code in Zuul that deals with Cookies, seems to use _getAll()_ and loop through them.

FWIW, as you said, the RFC does explicitly not allow multiple Cookies in the request..

https://tools.ietf.org/html/rfc6265#section-5.4
.."When the user agent generates an HTTP request, the user agent MUST
   NOT attach more than one Cookie header field...."

HTTP/2 allows multiple Cookie fields via HPACK (for better compression efficiency), but, does state that the HPACK decoder must concatenate all such fields into a single Header before passing it to a HTTP/1.1 context..

https://tools.ietf.org/html/rfc7540#section-8.1.2.5

**Pull request merged**